File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/screens/pages/sections/SectionsViewModel.java
package ru.usedesk.knowledgebase_gui.screens.pages.sections;

import androidx.annotation.NonNull;

import java.util.List;

import ru.usedesk.knowledgebase_gui.screens.common.DataViewModel;
import ru.usedesk.knowledgebase_gui.screens.common.ViewModelFactory;
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase;
import ru.usedesk.knowledgebase_sdk.external.entity.Section;

class SectionsViewModel extends DataViewModel<List<Section>> {

    private SectionsViewModel(@NonNull IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk) {
        loadData(usedeskKnowledgeBaseSdk.getSectionsRx());
    }

    static class Factory extends ViewModelFactory<SectionsViewModel> {
        private final IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk;

        public Factory(@NonNull IUsedeskKnowledgeBase usedeskKnowledgeBaseSdk) {
            this.usedeskKnowledgeBaseSdk = usedeskKnowledgeBaseSdk;
        }

        @NonNull
        @Override
        protected SectionsViewModel create() {
            return new SectionsViewModel(usedeskKnowledgeBaseSdk);
        }

        @NonNull
        @Override
        protected Class<SectionsViewModel> getClassType() {
            return SectionsViewModel.class;
        }
    }
}

File: knowledgebase-gui/src/main/java/ru/usedesk/knowledgebase_gui/internal/screens/pages/sections/SectionsViewModel.kt
package ru.usedesk.knowledgebase_gui.internal.screens.pages.sections

import ru.usedesk.knowledgebase_gui.internal.screens.common.DataViewModel
import ru.usedesk.knowledgebase_gui.internal.screens.common.ViewModelFactory
import ru.usedesk.knowledgebase_sdk.external.IUsedeskKnowledgeBase
import ru.usedesk.knowledgebase_sdk.external.entity.UsedeskSection

class SectionsViewModel private constructor(usedeskKnowledgeBaseSdk: IUsedeskKnowledgeBase) : DataViewModel<List<UsedeskSection?>?>() {
    internal class Factory(private val usedeskKnowledgeBaseSdk: IUsedeskKnowledgeBase) : ViewModelFactory<SectionsViewModel?>() {
        override fun create(): SectionsViewModel {
            return SectionsViewModel(usedeskKnowledgeBaseSdk)
        }

        override fun getClassType(): Class<SectionsViewModel?> {
            return SectionsViewModel::class.java
        }
    }

    init {
        loadData(usedeskKnowledgeBaseSdk.sectionsRx)
    }
}